### PR TITLE
Allow InternetStreams on LAN to search for .nfo and image files for videos

### PR DESCRIPTION
**Pull request by snaunton**
## Description

Allow Internet Streams on LAN to search for .nfo and image files for videos

Allow Internet Streams on LAN to search for .nfo and image files for videos

## Motivation and Context

Allow Internet Streams on LAN to search for .nfo and image files. There does not seem to be a reason to avoid this functionality. I'm sure I cannot be alone in using http to serve videos on my LAN as it is ubiquitous and fast, and clients do not need write access. For home, or any other videos which are not on scraper sites, no allow .nfo and thumb/poster images is a hindrance.

N/A

## How Has This Been Tested?

Have ran against my library at home which has some .nfo and images for videos which work with smb, but not http until this fix.

Kodi on linux (Arch)
Video, .nfo and images served via http on a Synology NAS.
Added Movie and TV Show "Web server directory (HTTP)" sources 

It doesn't appear to.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ X] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -3043,7 +3043,7 @@ bool CFileItem::SkipLocalArt() const
        || StringUtils::StartsWithNoCase(m_strPath, "newsmartplaylist://")
        || StringUtils::StartsWithNoCase(m_strPath, "newplaylist://")
        || m_bIsShareOrDrive
-       || IsInternetStream()
+       || (IsInternetStream() && !URIUtils::IsOnLAN(GetPath()))
```
Comment (FernetMenta) [inline xbmc/FileItem.cpp]: better use GetDynPath()

File: xbmc/video/tags/VideoInfoTagLoaderFactory.cpp [hunk 2]
```diff
@@ -23,7 +24,7 @@ IVideoInfoTagLoader* CVideoInfoTagLoaderFactory::CreateLoader(const CFileItem& i
                                                               bool forceRefresh)
 {
   // don't try to read tags for streams
-  if (item.IsInternetStream())
+  if (item.IsInternetStream() && !URIUtils::IsOnLAN(item.GetPath()))
```
Comment (FernetMenta) [inline xbmc/video/tags/VideoInfoTagLoaderFactory.cpp]: better use GetDynPath()

File: xbmc/FileItem.cpp (hunk 1 above)
Comment (snaunton) [reply xbmc/FileItem.cpp]: Have made change. Not sure why jenkins is failing. Builds and tests ok locally.

File: xbmc/video/tags/VideoInfoTagLoaderFactory.cpp (hunk 2 above)
Comment (snaunton) [reply xbmc/video/tags/VideoInfoTagLoaderFactory.cpp]: Have made change. Not sure why jenkins is failing. Builds and tests ok locally.

**Comment by Rechi** [comment]
@snaunton please squash your commits.

**Pull request closed without merging**